WalletWasabi.Tests/CommandInterpreterTests.cs

```diff
@@ -0,0 +1,160 @@
+
+			var runGui = await CommandInterpreter.ExecuteCommandsAsync(null, new string[0]);
+			Assert.True(runGui);
+		}
+
+		[Fact]
+		// When invoked with "help" (or "--help") it should NOT in GUI mode
+		public async Task InvokedWithHelpAsync()
+		{
+			var ms = new MemoryStream();
+			var output = new StreamWriter(ms, Encoding.ASCII, 1024, true);
+			CommandInterpreter.Configure(CreateMixCommand(), CreateFindPasswordCommand(), output, output);
+			var runGui = await CommandInterpreter.ExecuteCommandsAsync(null, new[]{ "help" });
+			output.Flush();
+			var consoleOutput = Encoding.ASCII.GetString(ms.ToArray());
+			Assert.Contains("Usage:", consoleOutput);
+			Assert.False(runGui);
+
+			ms.Flush();
+			CommandInterpreter.Configure(CreateMixCommand(), CreateFindPasswordCommand(), output, output);
+			runGui = await CommandInterpreter.ExecuteCommandsAsync(null, new[]{ "--help" });
+			output.Flush();
+			consoleOutput = Encoding.ASCII.GetString(ms.ToArray());
+			Assert.Contains("Usage:", consoleOutput);
+			Assert.False(runGui);
+		}
+
+		[Fact]
+		// When invoked with "--version" it should NOT in GUI mode
+		public async Task InvokedWithVersionAsync()
+		{
+			var ms = new MemoryStream();
+			var output = new StreamWriter(ms, Encoding.ASCII, 1024, true);
+			CommandInterpreter.Configure(CreateMixCommand(), CreateFindPasswordCommand(), output, output);
+			var runGui = await CommandInterpreter.ExecuteCommandsAsync(null, new[]{ "--version" });
+			output.Flush();
+			var consoleOutput = Encoding.ASCII.GetString(ms.ToArray());
+			Assert.Contains("Wasabi Client Version:", consoleOutput);
+			Assert.False(runGui);
+		}
+
+		[Fact]
+		// When invoked with "--datadir" only it should set the DataDir and launch the GUI
+		public async Task InvokedWithDataDirAsync()
+		{
+			var global = new WalletWasabi.Gui.Global();
+			var ms = new MemoryStream();
+			var output = new StreamWriter(ms, Encoding.ASCII, 1024, true);
+			CommandInterpreter.Configure(CreateMixCommand(), CreateFindPasswordCommand(), output, output);
+			var runGui = await CommandInterpreter.ExecuteCommandsAsync(global, new[]{ "--datadir", "expected/passed/datadir" });
+			output.Flush();
+			Assert.Equal("expected/passed/datadir", global.DataDir);
+			Assert.True(runGui);
+		}
+
+		[Fact]
+		// When invoked with "--datadir" only it should set the DataDir and launch the GUI
+		public async Task InvokedWithMixAsync()
+		{
+			var ms = new MemoryStream();
+			var output = new StreamWriter(ms, Encoding.ASCII, 1024, true);
+			var invoked = false;
+			var mixCommand = CreateMixCommand(); 
+			mixCommand.Run = (s)=> invoked = true;
+			CommandInterpreter.Configure(mixCommand, CreateFindPasswordCommand(), output, output);
+			var runGui = await CommandInterpreter.ExecuteCommandsAsync(null, new[]{ "mix" });
+			Assert.True(invoked);
+			Assert.False(runGui);
+
+			// Ensure backward compatibility with --mix
+			mixCommand = CreateMixCommand(); 
+			mixCommand.Run = (s)=> invoked = true;
```

Comment:
```suggestion
			mixCommand.Run = (s) => invoked = true;
```
Maybe the parenthesis around the s should be removed.